nasa/earthdata-search
# EDSC-3773: Refactor buildAccessMethods.js

# Overview

### What is the feature?

Refactored the buildAccessMethods.js file to pull out the 5 different logic blocks in order to make the file more readable and easier to modify (add new access methods in the future)

### What is the Solution?

- Moved out all the logic bocks in buildAccessMethods into their one files (1 for each logic block).
- Created tests for each of those logic blocks and updated the buildAccessMethods to test that those methods get called the correct number of times and with the correct params.

### What areas of the application does this impact?

List impacted areas.

# Testing

### Reproduction steps

- **Environment for testing: localhost**
- **Collection to test with: N/A**

1. go to [Test Collections](https://wiki.earthdata.nasa.gov/pages/viewpage.action?spaceKey=EDSC&title=Test+Collections) and pick out 1 collection for each accessMethod type.
EX: 
oPeNDAP: C1453188197-GES_DISC
Harmony: C1595422627-ASF
Echo Order: C179031504-LARC
Echo Order: C1200382306-CMR_ONLY (In SIT, testing new feature allowing for multiple Echo Order/ESI services to show up to be downloaded)
ESI: (The first good collection I found was in SIT C1200427107-EDF_DEV06)
Download: (any of the top 4)
2. navigate to locahost:8080/search?ee=prod
3. for each collection, add some granules and click download
4. Observe the downloads work correctly.

### Attachments

Please include relevant screenshots or files that would be helpful in reviewing and verifying this change.

# Checklist
